article/_posts/2020-08-14-IsolatingSourcesOfDisentanglementInVAEs.md

```diff
@@ -0,0 +1,120 @@
+---
+layout: review
+title: "Isolating Sources of Disentanglement in VAEs"
+tags: deep-learning autoencoder VAE
+author: "Nathan Painchaud"
+cite:
+    authors: "Ricky T. Q. Chen, Xuechen Li, Roger Grosse, David Duvenaud"
+    title:   "Isolating Sources of Disentanglement in VAEs"
+    venue:   "NeurIPS 2018"
+pdf: "https://arxiv.org/pdf/1802.04942.pdf"
+---
+
+
+# Highlights
+- Decomposition of the variational lower bound that explains the success of $$\beta$$-VAE in learning disentangled
+representations
+- Simple method based on weighted minibatches to estimate the aggregate posterior without additional hyperparameters
+(the aggregate posterior is required to weight specific terms in the decomposition of the ELBO)
+- New $$\beta$$-TCVAE (Total Correlation Variational Autoencoder), which can be used as a plug-in replacement for the
+$$\beta$$-VAE (and VAE for that matter)
+- New information-theoretic disentanglement metric, which is classifier-free and generalizable
+
+
+# Introduction
+The paper was written independently from a similar paper by Kim et Mnih[^1], and in the same vein propose a tweak on
+the $$\beta$$-VAE[^2] to further disentangle the learned representation without sacrificing the reconstruction,
+although the modifications proposed by both papers differ.
+
+Lets recall the objective of the $$\beta$$-VAE:
+
+$$
+\tag{1} \mathcal{L}_{\beta} = \frac{1}{N} \sum_{n=1}^{N} (\mathbb{E}_{q} [\log p(x_{n}|z)] - \beta KL(q(z|x_{n})~\|~p(z)))
+$$
+
+where $$\beta \gt 1$$ to encourage disentanglement. This is a more general formulation of the VAE, since we can obtain
+the original VAE objective by setting $$\beta = 1$$ in eq. 1.
+
+
+# Methods
+
+## ELBO TC-Decomposition
+Let's define $$q(z|n) = q(z|x_{n})$$ and $$q(z,n) = q(z|n) p(n) = q(z|n) \frac{1}{N}$$.
+The *aggregate posterior* is expressed as $$q(z) = \sum_{n=1}^{N} q(z|n) p(n)$$. The authors decompose the KL term
+like follows:
+
+$$
+\begin{aligned}
+\tag{2} \mathbb{E}_{p(n)} [KL(q(z|n)~\|~p(z))] = & \underbrace{KL(q(z,n)~\|~q(z) p(n))}_{\text{1. Index-Code MI}}
++ \underbrace{KL(q(z)~\|~\prod_{j} q(z_{j}))}_{\text{2. Total Correlation}} \\
+& + \underbrace{\sum_{j} KL(q(z_{j})~\|~p(z_{j}))}_{\text{3. Dimension-wise KL}}
+\end{aligned}
+$$
+
+The authors describe the purpose of each term:
+1. The *index-code mutual information* (MI) is the mutual information $$I_{q}(z;n)$$ between the data variable and
+latent variable;
+2. The *total correlation* (TC) is a generalization of mutual information to more than two random variables. A heavier
+penalty on this term is what encourages a disentangled representation;
+3. The *dimension-wise* KL prevents individual latent dimensions from deviating too far from their priors.
+
+## Training with Minibatch-Weighted Sampling
+The decomposed expression from eq. 2 requires to evaluate the aggregate posterior, which depends on the entire dataset
```

Comment:
_eq. 2_ -> We do not have equation numbering when writing equations with LaTeX, right?

Reply:
See my answer above.

Reply:
:+1:, sorry again, and thanks for the explanation.